File: dotnetcoresignalrclientjava/src/main/java/com/smartarmenia/dotnetcoresignalrclientjava/HubConnection.java
package com.smartarmenia.dotnetcoresignalrclientjava;

public interface HubConnection {
    void connect();

    void disconnect();

    boolean isConnected();

    void addListener(HubConnectionListener listener);

    void removeListener(HubConnectionListener listener);

    void subscribeToEvent(String eventName, HubEventListener eventListener);

    void unSubscribeFromEvent(String eventName, HubEventListener eventListener);

    void invoke(String event, Object... parameters);
}

File: dotnetcoresignalrclient/src/main/java/com/smartarmenia/dotnetcoresignalrclient/HubConnection.kt
package com.smartarmenia.dotnetcoresignalrclient

interface HubConnection {
    fun connect(authHeader: String? = null)
    fun disconnect()
    fun addListener(listener: HubConnectionListener)
    fun removeListener(listener: HubConnectionListener)
    fun subscribeToEvent(eventName: String, eventListener: HubEventListener)
    fun unSubscribeFromEvent(eventName: String, eventListener: HubEventListener)
    fun invoke(event: String, vararg parameters: Any)
}
